packages/rest/src/route.ts

```diff
@@ -482,6 +482,11 @@ function isUninstantiatedTemplateInterface(maybeInterface: Type): boolean {
   );
 }
 
+function isUninstantiatedTemplateOperation(maybeOperation: Type): boolean {
+  // Any operation statement with template parameters is inherently uninstantiated
+  return maybeOperation.kind === "Operation" && maybeOperation.node.templateParameters.length > 0;
```

Comment:
what is the difference? wouldn't the type also have `templateArguments`? would it not be a bug in the type creation instead? also should maybe add a test for that

Reply:
I was previously checking for type arguments when I should have been checking for type parameters.  The difference is that a signature definition has parameters but a signature *usage* has arguments.  This meant that base operation signatures were incorrectly being used as routes.

Reply:
yeah nvm, this is right, keep getting confused by the `templateParameters` vs `templateArguments`

Reply:
Me too! 😅   Added a test to verify that op templates are skipped correctly.

Reply:
Hmm, I'm still confused. Do instantiated operations not have the same node as the template they are instantiated from? We have a variety of these checks around. I think it's time to export something general from the compiler and use it.

Reply:
I filed #601

Reply:
Yeah, it'd definitely be nice to have a common function for this. 

An `OperationType`'s node is the node where the operation was declared, regardless of whether its reusing the signature of another operation, so it has its own unique node.

Reply:
Less obscurely, do we see such an OperationType when checking the rhs of `op lhs is rhs<arg>`, and in that case the shared helper would definitely need to be generalized to check for no arguments.

Reply:
yeah wondering if we should allow templated operation to be referenced. There is another issue about referencing operation making openapi3 emitter emit a confusing diagnostic but in this case it feels like we shouldn't maybe allow inline operation instantiation.

I am down to add a check to prevent instantiating an operation outside of `op foo is ...`